src/test/java/BaseTest.java

```diff
@@ -1,11 +1,268 @@
+        actions.doubleClick(playlistElement).perform();
+    }
+
+    //Play song helper functions
+    public void chooseAllSongList() {
+        wait.until(ExpectedConditions.invisibilityOfElementLocated(By.cssSelector(".overlay.loading")));
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("li a.songs"))).click();
+    }
+
+    public void contextClickFirstSong() {
+        WebElement firstSongElement = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".all-songs tr.song-item:nth-child(1)")));
+        actions.contextClick(firstSongElement).perform();
+    }
+
+    public void choosePlayOption() {
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".song-menu li.playback"))).click();
+    }
+
+    //Hover helper function
+    public WebElement hoverPlay() {
+        WebElement play = driver.findElement(By.cssSelector("[data-testid='play-btn']"));
+        actions.moveToElement(play).perform();
+        return driver.findElement(By.cssSelector("[data-testid='play-btn']"));
+    }
+
+    //Count Songs Helper Functions
+    public void choosePlayListByName(String playlistName){
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//a[contains(text(),'" + playlistName + "')]"))).click();
+    }
+
+    public int countSongs(){
+        return driver.findElements(By.cssSelector("section#playlistWrapper td.title")).size();
+    }
+    public String getPlaylistDetails(){
+        return driver.findElement(By.cssSelector("span.meta.text-secondary span.meta")).getText();
+    }
+
+    public void displayAllSong(){
+        List<WebElement> songList = driver.findElements(By.cssSelector("section#playlistWrapper td.title"));
+        System.out.println("Number of Songs Found:" + countSongs());
+            for (WebElement e: songList){
+                System.out.println(e.getText());
+            }
+    }
+
+    //Helper function for Rename
+    String newPlaylistName = "new name";
+    public void doubleClickPlaylist(){
+        WebElement playlistElement = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".playlist:nth-child(3)")));
+        actions.doubleClick(playlistElement).perform();
+    }
+
+    public void enterNewPlaylistName() {
+        WebElement playlistInputField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[name='name']")));
+        playlistInputField.sendKeys(Keys.chord(Keys.COMMAND, "A", Keys.BACK_SPACE));
+        playlistInputField.sendKeys(newPlaylistName);
+        playlistInputField.sendKeys(Keys.ENTER);
+    }
+
+    public  boolean doesPlaylistExist(){
+        WebElement playlistElement = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//a[text()='" + newPlaylistName + "']")));
+        return playlistElement.isDisplayed();
+    }
```

Comment:
Page Object Model (POM): recommended to associate the methods with the corresponding page class where they are relevant. In the POM, each page or section of a web application is represented by a separate class. These classes encapsulate the elements and actions specific to that page or section.

**Move these methods to the corresponding class according to the POM.**

For example 

![image](https://github.com/SupattraTang/A47-QA-Automation-Framework/assets/104761482/7df20955-4da4-421a-b07f-ba673b43298a)